pkg/ext-proc/backend/llmserverpool_controller.go

```diff
@@ -0,0 +1,58 @@
+package backend
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"inference.networking.x-k8s.io/llm-instance-gateway/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	controllerNamePrefix = "instance-gateway-"
+)
+
+// LLMServerPoolController is the controller implementation for Instance Gateway resources
+type LLMServerPoolController struct {
+	client.Client
+	Scheme         *runtime.Scheme
+	Record         record.EventRecorder
+	ServerPoolName string
+	Namespace      string
+	Datastore      *K8sDatastore
+	Port           int
+	Zone           string
+}
+
+func (c *LLMServerPoolController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if req.NamespacedName.Name != c.ServerPoolName && req.NamespacedName.Namespace != c.Namespace {
+		return ctrl.Result{}, nil
+	}
+	klog.V(1).Info("reconciling LLMServerPool", req.NamespacedName)
+
+	serverPool := &v1alpha1.LLMServerPool{}
```

Comment:
aren't we supposed to create a Service here?

Reply:
I realized that it would be confusing to have the EPP reconcile and create an object on the LLMServerPool's behalf, in addition to whatever controller created the EPP. Instead, the implementation of the LLMServerPool controller should create the service and pass the service name in on startup. I think centralizing the object creation/control will make for a cleaner interface/experience

Reply:
Leaving this comment unresolved for posterity. But discussed further offline. And it sounds like we are in agreement here